extension/app/ts/components/subcomponents/address.tsx

```diff
@@ -20,48 +20,19 @@ export function findAddressInfo(addressToFind: bigint, addressInfos: readonly Ad
 	}
 }
 
-export type BigAddressParamsWithChainSelector = {
-	address: bigint
-	title: string | undefined
-	chainId: bigint
-	changeActiveChain: (chainId: bigint) => void
-}
-
-export function BigAddressWithChainSelector(param: BigAddressParamsWithChainSelector) {
-	const title = param.title === undefined ? ethers.utils.getAddress(addressString(param.address)): param.title
-	const addrString = ethers.utils.getAddress(addressString(param.address))
-
-	return <div class = 'media'>
-		<div class = 'media-left'>
-			<CopyToClipboard content = { addrString } copyMessage = 'Address copied!'>
-				<figure class = 'image noselect nopointer'>
-					<Blockie seed = { addressString(param.address).toLowerCase() } size = { 8 } scale = { 5 } />
-				</figure>
-			</CopyToClipboard>
-		</div>
-
-		<div class = 'media-content' style = 'overflow: visible;'>
-			<ChainSelector currentChain = { param.chainId } changeChain = { (chainId: bigint) => { param.changeActiveChain(chainId) } }/>
-			<CopyToClipboard content = { addrString } copyMessage = 'Address copied!'>
-				<p class = 'subtitle is-5 noselect nopointer'>
-					{ title }
-				</p>
-			</CopyToClipboard>
-		</div>
-	</div>
-}
-
 export type BigAddressParams = {
 	address: bigint
 	title?: string
 	subtitle?: string
 	noCopying?: boolean
+	renameAddressCallBack: RenameAddressCallBack | undefined
```

Comment:
```suggestion
	readonly address: bigint
	readonly title?: string
	readonly subtitle?: string
	readonly noCopying?: boolean
	readonly renameAddressCallBack: RenameAddressCallBack | undefined
```
Generally speaking, object properties should be marked as readonly.  This makes auditing the code *way* easier, and often also encourages good design patterns as well (like no shared objects between different parts of an application).

Reply:
(with both pieces of feedback included)
```suggestion
	readonly address: bigint
	readonly title?: string
	readonly subtitle?: string
	readonly noCopying?: boolean
	readonly renameAddressCallBack?: RenameAddressCallBack
```

Reply:
Same comment for all of the other places with this pattern (both in this PR and throughout the codebase).